scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -259,6 +267,46 @@ public void run() {
         return true;
     }
 
+    /**
+     * Create a new Credential object containing users' 3rd Party Credentials.
+     *
+     * @param creds credentials for specific user
+     * @return in case of success new object containing the 3rd party credentials used to create bindings
+     * at clean script
+     */
+    Credentials addThirdPartyCredentials(Credentials creds) throws KeyException, IllegalAccessException {
+        //retrieve scheduler key pair
+        String privateKeyPath = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PRIVKEY_PATH.getValueAsString());
+        String publicKeyPath = PASchedulerProperties.getAbsolutePath(PASchedulerProperties.SCHEDULER_AUTH_PUBKEY_PATH.getValueAsString());
+
+        //get keys from task
+        PrivateKey privateKey = Credentials.getPrivateKey(privateKeyPath);
+        PublicKey publicKey = Credentials.getPublicKey(publicKeyPath);
+
+        //retrieve the current creData from task
+        CredData credData = creds.decrypt(privateKey);
+
+        //retrive database to get third party credentials from
+        SchedulerDBManager dbManager = getInfrastructure().getDBManager();
+        if (dbManager != null) {
+            Map<String, HybridEncryptedData> thirdPartyCredentials = null;
```

Comment:
no need to assign thirdPartyCredentials = null;
Directly do 
Map<String, HybridEncryptedData> thirdPartyCredentials = dbManager.thirdPartyCredentialsMap(credData.getLogin());

Reply:
ok i agree.